terraform/api_gateway.tf.json.template.py

```diff
@@ -196,12 +202,104 @@ def for_domain(cls, domain):
+                            'name': 'AWS-CommonRuleSet',
+                            'override_action': {
+                                'none': {}
+                            },
+                            'statement': {
+                                'managed_rule_group_statement': {
+                                    'name': 'AWSManagedRulesCommonRuleSet',
+                                    'vendor_name': 'AWS'
+                                }
+                            },
+                            'visibility_config': {
+                                'metric_name': 'AWS-CommonRuleSet',
+                                'sampled_requests_enabled': True,
+                                'cloudwatch_metrics_enabled': True
+                            }
+                        },
+                        {
+                            'priority': 3,
+                            'name': 'AWS-AmazonIpReputationList',
+                            'override_action': {
+                                'none': {}
+                            },
+                            'statement': {
+                                'managed_rule_group_statement': {
+                                    'name': 'AWSManagedRulesAmazonIpReputationList',
+                                    'vendor_name': 'AWS'
+                                }
+                            },
+                            'visibility_config': {
+                                'metric_name': 'AWS-AmazonIpReputationList',
+                                'sampled_requests_enabled': True,
+                                'cloudwatch_metrics_enabled': True
+                            }
+                        },
+                        {
+                            'priority': 4,
+                            'name': 'AWS-UnixRuleSet',
+                            'override_action': {
+                                'none': {}
+                            },
+                            'statement': {
+                                'managed_rule_group_statement': {
+                                    'name': 'AWSManagedRulesUnixRuleSet',
+                                    'vendor_name': 'AWS'
+                                }
+                            },
+                            'visibility_config': {
+                                'metric_name': 'AWS-UnixRuleSet',
+                                'sampled_requests_enabled': True,
+                                'cloudwatch_metrics_enabled': True
+                            }
+                        },
                     ],
                     'scope': 'REGIONAL',
                     'visibility_config': {
                         'cloudwatch_metrics_enabled': True,
                         'metric_name': 'WebACL',
                         'sampled_requests_enabled': True,
+                    },
+                    'lifecycle': {
+                        'create_before_destroy': True
```

Comment:
This is usually a bad smell. Please explain.

Reply:
Ah. The `create_before_destroy` argument was necessary to get past a `couldn’t perform the operation because your resource is being used by another resource` issue I was running into during deployment. This however was before it was decided to move the `aws_wafv2_ip_set` to the shared component, and add the `ignore_changes` lifecycle to the IP set.

I've confirmed with my personal deployment the `create_before_destroy` is no longer needed, and added a fixup commit to remove it.

Reply:
Please describe how exactly you confirmed the fact that create_before_destroy is not needed. Just redeploying without it is usually not sufficient because the resource already exists and will not be touched.

Reply:
The steps I followed were to:
- Deployed `develop` to my deployment
- Using the AWS console, manually created an IP set named `blocked` in the dev account
- Deployed my feature branch (with `create_before_destroy` removed) to my deployment, confirmed WAF rules were created
- Deployed `develop` to my deployment, confirmed WAF rules were removed
- Manually deleted the `blocked` IP set.

Reply:
The resource we're discussing here is in the `shared` component. I'm not sure that redeploying a personal deployment is conclusive evidence about a resource that's not even in that deployment.

If there already is an IP set called `blocked` in any of our AWS accounts (please check), we need to make sure that we handle that case in the upgrading instructions, by instructing the operator to import or delete it. Probably the latter.

Be sure to test the complete deployment scenario. This test must involve two components, `shared` and the main component of a personal deployment.